### DISCUSSION TITLE
Is it possible to customize mount point with `volume`?

### DISCUSSION THREAD

**Discussion by charnould** [basecamp/kamal#340] [Q&A]
Currently, if your `mrsk.yaml` contains...

```
volumes:
  - 'database:/app/database'
```

... it creates a `volume`:  
code-available at `app/database`,  
but located at `/var/lib/docker/volumes/database/_data` (default `Docker` location for `volume`).

Using Hetzner, I'd like to use their OWN [Volume ](https://docs.hetzner.com/cloud/volumes/), kind of mini-S3.
After having configured one, this Hetzner Volume is available at `/mnt/...`

Is it possible with `mrsk` to configure the mount point?
In Docker-word it should be (?) a `bind`-mount (?)

Thanks.

**Comment by ncompute** [comment]
@charnould  - Got this working but it could  be improved.

On your host (replace path based on your requirements):
```sh
$ mkdir /root/data && chmod 777 /root/data
```

Then in your `deploy.yml` file:
```yml
volumes:
  - "/root/data:/rails/storage/db"
```


Note: Looks like this will only work with **absolute** host paths:
https://github.com/basecamp/kamal/blob/83a2d52ff41eeb9c3240951316792c15223e8c98/lib/kamal/configuration/volume.rb#L15-L21

**Comment by abratashov** [comment]
Yes, it works! But I'm curious why this setting isn't present in the default generated `deploy.yml` for the folder `storage`:
```sh
$ sudo mkdir /data && sudo chmod 777 /data
```
```yml
volumes:
  - "/data/storage:/rails/storage"
```

**Reply by acidtib** [reply]
Kamal is a bit framework-agnostic. Not every deployment is going to be for Rails. That is why the volume definition is not set as default for /rails/storage on kamal init.

**Reply by abratashov** [reply]
Oh, indeed! I totally forgot about it :)
But in any case, in the future hypothetical `kamal-rails` gem, the comments about volumes are vital for any web app to upload/download data backup, hope it will be added to the default `deploy.yml`:
```yml
# ...
# To restore or make a data backup, you can link your host's folder to the app of docker container:
# volumes:
#   - "/data/storage:/rails/storage"
# ...
```
or possibly something like this (I'm not sure if it works because I haven't tested it yet):
```yml
# accessories:
#   web:
#     directories:
#       - data/storage:/rails/storage
```
